pallets/settlement/src/lib.rs

```diff
@@ -927,25 +942,19 @@ impl<T: Config> Module<T> {
         // Ensure venue exists & sender is its creator.
         Self::venue_for_management(venue_id, did)?;
 
-        // Prepare data to store in storage.
+        // Create a list of unique counter parties involved in the instruction.
         let mut counter_parties = BTreeSet::new();
-        let mut tickers = BTreeSet::new();
-        // This is done to create a list of unique CP and tickers involved in the instruction.
         for leg in &legs {
             ensure!(leg.from != leg.to, Error::<T>::SameSenderReceiver);
-            counter_parties.insert(leg.from);
-            counter_parties.insert(leg.to);
-            tickers.insert(leg.asset);
-        }
-
-        // Check if the venue has required permissions from token owners.
-        for ticker in &tickers {
-            if Self::venue_filtering(ticker) {
+            // Check if the venue has required permissions from token owners.
+            if Self::venue_filtering(leg.asset) {
                 ensure!(
-                    Self::venue_allow_list(ticker, venue_id),
+                    Self::venue_allow_list(leg.asset, venue_id),
```

Comment:
N.B. this change can theoretically increase the number of checks and database reads, although it should be cached behind the scenes, but this could be a pessimization. But let's try this anyways.